cmake-modules/AzureBuildTargetForCI.cmake

```diff
@@ -4,8 +4,21 @@
 # Defines utility functions to create build targets for CI.
 #
 
-macro(create_per_service_target_build target service)
+macro(create_per_service_target_build target service runAsSample)
```

Comment:
> is too easy to miss 

How come?  Macro will throw if nothing is passed as 3rd argument.

I thought about a new method..  But, usually the projects are copy/pasted, and changing `true/false` is easier than looking for the name of the other macro.  Also to promote any binary (not only a sample), to be run on CI, I would like to have this sort of switch `ON/OFF` instead of a 2 macros. What do you think?

Reply:
In my experience, terminal boolean parameters are (a) easy to miss, and (b) easy to misuse. 

Can you immediately explain the difference between:

```
create_per_service_target_build(keyvault-credentials keyvault, true)
```

```
create_per_service_target_build(keyvault-credentials keyvault, false)
```
?

If you were visually scanning through the code, would you pick up on the fact that the terminal boolean is different?

Contrast that with:
```
create_per_service_target_build_for_sample(keyvault-credentials keyvault)
```

```
create_per_service_target_build(keyvault-credentials keyvault)
```

Using a separate macro name makes it extremely clear that one of these forms is for samples and the other form is for things which aren't samples.

And if you cut and pasted this code, the odds of you accidentally copying `create_per_service_target_build(keyvault-credentials keyvault, false)` into a sample directory (or vice versa) is extremely high. When scanning through code looking for differences, looking for the `_for_sample` suffix is easy to see - the terminal "true" is not.

Reply:
still think that it's kind of easy. But in the end it's about the same. And we don't have cmake guidelines documented :)

I have updated as you suggested anyway. Thank you for reviewing